sources/rules/import-absolutes.ts

```diff
@@ -80,10 +95,12 @@ const rule: Rule.RuleModule = {
 
     return {
       ImportDeclaration (node) {
-        if (typeof node.source.value !== `string`)
+        const importPath = node.source.value;
+
+        if (typeof importPath !== `string`)
           return;
 
-        if (!node.source.value.match(/^\.{0,2}\//))
+        if (!importPath.match(/^\.{0,2}\//) || keepRelative(importPath))
           return;
 
         reportExpectedAbsoluteImportError(node);
```

Comment:
I think the check should be done within `reportExpectedAbsoluteImportError`. This has two benefits:

- First we get to run the regex against the normalized relative path[1], to avoid issues if a relative import is somehow written `./foo`, `.//foo`, `././foo`, etc.
- We could then also autofix "incorrect" absolute paths into relative ones (useful if, say, TS auto-import mistakenly adds the imports as absolute rather than relative).

[1] To get the normalized relative path in `reportExpectedAbsoluteImportError`:
```ts
const normalizedRelative = path.relative(path.dirname(sourcePath), targetPath) || `.`;
```

Reply:
> We could then also autofix "incorrect" absolute paths into relative ones (useful if, say, TS auto-import mistakenly adds the imports as absolute rather than relative).

That's something I've been wondering about. At the moment the option is about which relative imports should be allowed to remain so; it doesn't force some imports to be relative (eg imports from the same folder). Do you think that should be the case (and then either the option should be renamed `forceRelative`, or there should be another option to specify that behaviour)

Reply:
Yeah, I think it's better to enforce relative/absolute, this way it really makes it the definite "prettier for imports". Renaming it is a good idea. Perhaps `preferRelative`?

Reply:
I've been thinking about it further - right now the rule is about turning any relative import into an absolute one, with some possible exception (added in this PR).

The rule workflow would be:

For any import:
- if absolute and matching `preferRelative`, turn into relative import
- if relative and not matching `preferRelative`, turn into absolute import

It seems somewhat of a departure from the rule's original intent, and I'm a bit concerned about specifying a pattern that makes sense in both absolute -> relative and relative -> relative contexts?

In any case I've implemented your first suggestion. Let me know what you think!